src/util.ts

```diff
@@ -32,13 +32,13 @@ export function isCustomElementDecorator(node: DecoratorNode): boolean {
  * Determines if a node is an element class or not.
  *
  * @param {Rule.RuleContext} context ESLint rule context
- * @param {ESTree.Node} node Node to test
+ * @param {ESTree.Class} node Node to test
  * @param {AST.Token=} jsdoc JSDoc to parse
  * @return {boolean}
  */
 export function isCustomElement(
   context: Rule.RuleContext,
-  node: ESTree.Node,
+  node: ESTree.Class,
   jsdoc?: AST.Token | null
 ): node is ESTree.Class {
```

Comment:
this and `isNativeCustomElement` should both probably just return bool now, since they were acting as type guards before but no need anymore as you pass a class in